### PR TITLE
New fish

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
Content "New fish"

#### Purpose of change
I like fish

#### Describe the solution

**NEW SPECIES**
- [x] Burbot
- [x] Yellow Bullhead
- [x] Longnose Gar
- [x] Lake Sturgeon
- [x] Shortnose Sturgeon
- [x] Round Whitefish
- [x] American Pickerel
- [x] Goldfish

**PROPER IMPLEMENTATION**
- [x] Add eggs and make them hatch
- [x] Make new fish spawn
- [x] Make eggs usable in the requirement group (except Gar)
- [x] Double-check fish sizes they copy from
- [x] Double-check descriptions to make sure you're not copying any
- [x] Double-check if the new fish have ``looks_like`` where applicable
- [x] Add new words to the dictionary

#### Describe alternatives you've considered
Initially I also wanted to add the Splake and the Amur Carp, as both are relatively common feral fish in the region the game takes place in. However, I had issues finding information about their size and so I opted against doing that.

#### Additional context
**CALCULATIONS**
Same as the last time with #61614
- Sturgeon, Burbot, Bullhead and Pickerel divided by 6
- Gar divided by 7
- Whitefish divided by 5
- Goldfish divided by 4

<details>
<summary>Burbot:</summary>

https://www.fishbase.se/summary/Lota-lota.html

https://animaldiversity.org/accounts/Lota_lota/

</details>
<details>
<summary>Yellow Bullhead:</summary>

https://www.fishbase.se/summary/Ameiurus-natalis.html

https://animaldiversity.org/accounts/Ameiurus_natalis/

</details>
<details>
<summary>Longnose Gar:</summary>

https://www.fishbase.se/summary/Lepisosteus-osseus.html

https://animaldiversity.org/accounts/Lepisosteus_osseus/

</details>
<details>
<summary>Lake Sturgeon:</summary>

https://www.fishbase.se/summary/Acipenser-fulvescens.html

https://animaldiversity.org/accounts/Acipenser_fulvescens/

</details>
<details>
<summary>Shortnose Sturgeon:</summary>

https://www.fishbase.se/summary/Acipenser-brevirostrum.html

</details>
<details>
<summary>Round Whitefish:</summary>

https://www.fishbase.se/summary/Prosopium-cylindraceum.html

</details>
<details>
<summary>American Pickerel:</summary>

https://animaldiversity.org/accounts/Esox_americanus/

</details>
<details>
<summary>Goldfish:</summary>

https://animaldiversity.org/accounts/Carassius_auratus/

</details>

**Post scriptum**
The spawning implementation is pretty half-assed, but I gotta take a closer look at all the fish monstergroups anyway. I intend to clean it up in a followup PR

### CODE REVIEW

File: data/json/monsters/fish.json [hunk 1]
```diff
@@ -660,6 +694,23 @@
     "upgrades": { "half_life": 14, "into": "mon_fish_pickerel" },
     "name": { "str_sp": "fry" }
   },
+  {
+    "id": "mon_fish_pickerel_american",
+    "type": "MONSTER",
+    "copy-from": "mon_fish_tiny",
+    "name": { "str": "american pickerel" },
+    "description": "An american pickerel.  It looks like a pike, but way smaller.",
```
Comment (TheShadowFerret) [inline data/json/monsters/fish.json:702]:
```suggestion
    "name": { "str": "American pickerel" },
    "description": "An American pickerel.  It looks like a pike, but way smaller.",
```

Comment (Karol1223) [reply data/json/monsters/fish.json:702]: Good catch! Forgot about that!

**Comment by Karol1223** [comment]
I'll point this out here right now - "cods" is a real word if used to refer to multiple TYPES of codfish. I'm not as stupid as I look, just fish and grammar don't really get along well.

File: data/json/monsters/fish.json [hunk 2]
```diff
@@ -748,6 +819,76 @@
...
+    "upgrades": { "half_life": 14, "into": "mon_fish_sturgeon_lake" },
+    "name": { "str_sp": "fry" }
+  },
+  {
+    "id": "mon_fish_sturgeon_shortnose",
+    "type": "MONSTER",
+    "copy-from": "mon_fish_small",
+    "name": { "str": "shortnose sturgeon" },
+    "looks_like": "mon_fish_sturgeon_lake",
+    "description": "While the shortnose sturgeon it is still large, it is nothing compared to its cousin - the lake sturgeon.  What really sets it apart, however, is its signature shorter snout.",
```
Comment (harakka) [inline data/json/monsters/fish.json:880]:
```suggestion
    "description": "While the shortnose sturgeon is still large, it is nothing compared to its cousin - the lake sturgeon.  What really sets it apart, however, is its signature shorter snout.",
```

File: data/json/monsters/fish.json [hunk 3]
```diff
@@ -660,6 +696,23 @@
     "upgrades": { "half_life": 14, "into": "mon_fish_pickerel" },
     "name": { "str_sp": "fry" }
   },
+  {
+    "id": "mon_fish_pickerel_american",
+    "type": "MONSTER",
+    "copy-from": "mon_fish_tiny",
+    "name": { "str": "American pickerel" },
+    "description": "An American pickerel.  It looks like a pike, but unlike its chain cousin, this one is way smaller.",
```
Comment (harakka) [inline data/json/monsters/fish.json:704]:
```suggestion
    "description": "An American pickerel.  It looks like a pike, but unlike its chain pickerel cousin, this one is way smaller.",
```
Reads better to people not familiar with the fish species

File: data/json/monsters/fish.json [hunk 4]
```diff
@@ -731,6 +784,24 @@
     "upgrades": { "half_life": 14, "into": "mon_fish_grass_carp" },
     "name": { "str_sp": "fry" }
   },
+  {
+    "id": "mon_fish_goldfish",
+    "type": "MONSTER",
+    "copy-from": "mon_fish_tiny",
+    "name": { "str_sp": "goldfish" },
+    "looks_like": "mon_fish_carp",
+    "description": "Yes, this is that common pet fish.  Though, given its size and the fact that it's still alive you can take a wild guess that this one is feral.",
```
Comment (harakka) [inline data/json/monsters/fish.json:793]:
```suggestion
    "description": "Yes, this is that common pet fish.  Though, given its size and the fact that it's still alive, you can take a wild guess that this one is feral.",
```

**Review by harakka — CHANGES_REQUESTED** [review]
Minor copyedit suggestions

**Comment by Theawesomeboophis** [comment]
I would love to be able to tame certain types of fish in the future with fishfood and put them in a tank, I think my survivor being able to have a pet goldfish or other type of fish would be really cool, in vanilla or tamable wildlife depending on consensus.

**Pull request merged**